h/accounts/services.py

```diff
@@ -115,16 +115,36 @@ def login(self, username_or_email, password):
 
         return None
 
-    def annotation_count(self, userid):
+    def public_annotation_count(self, userid):
         """
-        Return the count of shared annotations for this user.
+        Return the count of public annotations for this user.
         """
 
         return (
             self.session.query(Annotation)
             .filter_by(userid=userid, groupid='__world__',  shared=True)
             .count())
 
+    def all_groups_annotation_count(self, userid):
+        """
+        Return the count of shared annotations this user has made in groups.
+        """
+
+        return (
+            self.session.query(Annotation)
+            .filter_by(userid=userid, shared=True)
+            .count())
+
+    def private_annotation_count(self, userid):
```

Comment:
Given that each of these methods just queries the Annotation table, it seems to me that there's no particular reason these need to stay on the `UserService`, and it would be clearer if there was a separate `StatsService` for this.

Second, every round-trip to the database adds latency, so if we can avoid doing three, that would be good. Here's a quick sketch of what doing this in one query would look like:

```python
import sqlalchemy as sa
...

annotations = session.query(m.Annotation).filter_by(userid=userid).subquery()
grouping = sa.case([
    (sa.not_(annotations.c.shared), 'private'), 
    (annotations.c.groupid == '__world__', 'public'),
], else_='group')
results = session.query(sa.func.count(annotations.c.id), grouping).group_by(grouping).all()
```